osu.Game.Tests/Visual/Settings/TestSceneSettingsNumberBox.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Testing;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Overlays.Settings;
+
+namespace osu.Game.Tests.Visual.Settings
+{
+    public class TestSceneSettingsNumberBox : OsuTestScene
+    {
+        private SettingsNumberBox numberBox;
+        private OsuTextBox textBox;
+
+        [SetUpSteps]
+        public void SetUpSteps()
+        {
+            AddStep("create number box", () => Child = numberBox = new SettingsNumberBox());
+            AddStep("get inner text box", () => textBox = numberBox.ChildrenOfType<OsuTextBox>().Single());
+        }
+
+        [Test]
+        public void TestLargeInteger()
+        {
+            AddStep("set current to 1,000,000,000", () => numberBox.Current.Value = 1_000_000_000);
+            AddAssert("text box text is correct", () => textBox.Text == "1000000000");
+        }
+
+        [Test]
+        public void TestUserInput()
+        {
+            inputText("42");
+            currentValueIs(42);
+            currentTextIs("42");
+
+            inputText(string.Empty);
+            currentValueIs(null);
+            currentTextIs(string.Empty);
+
+            inputText("555");
+            currentValueIs(555);
+            currentTextIs("555");
+
+            inputText("-4444");
+            // attempting to input the minus will raise an input error, the rest will pass through fine.
+            currentValueIs(4444);
+            currentTextIs("4444");
+
+            // checking the upper bound.
+            inputText(int.MaxValue.ToString());
+            currentValueIs(int.MaxValue);
+            currentTextIs(int.MaxValue.ToString());
+
+            inputText((long)int.MaxValue + 1.ToString());
```

Comment:
```suggestion
            inputText((long)int.MaxValue + "1");
```

I think I'd prefer this unless there's a reason to write it otherwise (and github linting probably will too 😄 ).

Reply:
Hmm, that weakens the test a bit. I specifically wanted the first smallest value that overflows `int`.

Would it be better if I added parens to clarify precedence, or just hard coded the desired value?

Reply:
Okay if there's some precedence thing going on here then it 1000% needs parens. Because I can't/couldn't/won't see that.

Reply:
Have rewritten to something more legible, hopefully.